### ISSUE TITLE
integer overflow safety wrong source location

### ISSUE THREAD

**Issue by nektro** [ziglang/zig#13537]
### Zig Version

0.10.0

### Steps to Reproduce and Observed Behavior

```zig
test {
    var a: u8 = 0;
    var b = a - 1;

    _ = b;
}
```

```
Test [1/1] test_0... thread 2368606 panic: integer overflow
./test.zig:3:5: 0x211674 in test_0 (test)
    var b = a - 1;
    ^
/home/meghan/.local/share/zig/lib/test_runner.zig:63:28: 0x212c88 in main (test)
        } else test_fn.func();
                           ^
/home/meghan/.local/share/zig/lib/std/start.zig:596:22: 0x211f6b in posixCallMainAndExit (test)
            root.main();
                     ^
/home/meghan/.local/share/zig/lib/std/start.zig:368:5: 0x211a31 in _start (test)
    @call(.{ .modifier = .never_inline }, posixCallMainAndExit, .{});
    ^
error: the following test command crashed:
/home/meghan/zig-cache/o/fd7d7b7b0c9fad89e46921c0f7654494/test
```

### Expected Behavior

the safety to point to the operator that failed

```
thread 2368646 panic: integer overflow
./test.zig:3:5: 0x211674 in test_0 (test)
    var b = a - 1;
              ^
```

**Comment by Vexu** [comment]
Fixed by 25c850642190bf9939790b872a3657410ec4d19f

**Comment by nektro** [comment]
can confirm
